core/base/src/TString.cxx

```diff
@@ -238,11 +246,20 @@ TString::~TString()
 
 char *TString::Init(Ssiz_t capacity, Ssiz_t nchar)
 {
+   if (capacity < 0) {
+      Error("TString::Init", "Negative length!");
+      capacity = 0;
+   }
+   if (nchar < 0) {
+      Error("*TString::Init", "Negative length!");
+      nchar = 0;
+   }
+   if (nchar > capacity) {
+      nchar = capacity;
+   }
    if (capacity > MaxSize()) {
       Error("TString::Init", "capacity too large (%d, max = %d)", capacity, MaxSize());
       capacity = MaxSize();
-      if (nchar > capacity)
-         nchar = capacity;
```

Comment:
why did you move this code out of the outer `if`?

Reply:
I'm not sure 😅 . Should I put it back into the capacity > MaxSize() if-statement?

Reply:
yes please :smile: